02_cancer_type_classification/nbconverted/plot_univariate_fs_results.py

```diff
@@ -0,0 +1,377 @@
+# results files exist)
+cancer_types = None
+
+# these are the cancer types we use for TP53 (visualizing all of them gets a bit large/noisy)
+# cancer_types = [
+#     'BLCA',
+#     'BRCA',
+#     'COAD',
+#     'LGG',
+#     'LUAD',
+#     'SARC',
+#     'SKCM',
+#     'STAD',
+#     'UCEC'
+# ]
+
+
+# In[9]:
+
+
+sns.set({'figure.figsize': (18, 6)})
+sns.set_context('notebook')
+
+fig, axarr = plt.subplots(1, 3)
+
+dfs_to_plot = [
+    single_cancer_compare_df,
+    pancancer_compare_df,
+    pancancer_only_compare_df
+]
+
+names_to_plot = [
+    'Train single-cancer',
+    'Train pan-cancer',
+    'Train all other cancer types'
+]
+
+fs_method_order = [
+    'mad_250',
+    'mad_1000',
+    'pancan_f_test',
+    'median_f_test',
+    'mad_f_test'
+]
+
+
+for ix, compare_df in enumerate(dfs_to_plot):
+    ax = axarr[ix]
+    # averaged over cancer types
+    if cancer_types is None:
+        plot_df = compare_df[(compare_df.identifier == gene)]
+    else:
+        plot_df = compare_df[(compare_df.identifier == gene) &
+                             (compare_df.holdout_cancer_type.isin(cancer_types))]
+    sns.boxplot(data=plot_df, x='fs_method', y='delta_aupr',
+                order=fs_method_order, ax=ax)
+    ax.set_title(names_to_plot[ix])
+    ax.set_xlabel('Feature selection method')
+    ax.set_ylim(-0.2, 1)
+    
+plt.suptitle('{}, averaged over all holdout cancer types'.format(gene))
+plt.tight_layout()
+
+
+# In[10]:
+
+
+# these are "non-carcinoma" cancer types in TCGA
+# we anticipate that these will be the hardest to transfer models to (since
+# most of TCGA, and thus most of our training set, is generally carcinomas),
+# so it's also valuable to look at results filtered only to these
+non_carcinomas = [
+    'GBM',
+    'LGG',
+    'SARC',
+    'SKCM',
+    'TGCT'
+]
+
+
+# In[11]:
+
+
+sns.set({'figure.figsize': (18, 6)})
+sns.set_context('notebook')
+
+fig, axarr = plt.subplots(1, 3)
+
+dfs_to_plot = [
+    single_cancer_compare_df,
+    pancancer_compare_df,
+    pancancer_only_compare_df
+]
+
+names_to_plot = [
+    'Train single-cancer',
+    'Train pan-cancer',
+    'Train all other cancer types'
+]
+
+
+for ix, compare_df in enumerate(dfs_to_plot):
+    ax = axarr[ix]
+    # averaged over non-carcinoma cancer types
+    plot_df = compare_df[(compare_df.identifier == gene) &
+                         (compare_df.holdout_cancer_type.isin(non_carcinomas))]
+    if ix == 0:
+        # look at which cancer types are actually present in dataset
+        print(plot_df.holdout_cancer_type.unique(), file=sys.stderr)
+    sns.boxplot(data=plot_df, x='fs_method', y='delta_aupr',
+                order=fs_method_order, ax=ax)
+    ax.set_title(names_to_plot[ix])
+    ax.set_xlabel('Feature selection method')
+    ax.set_ylim(-0.2, 1)
+    
+plt.suptitle('{}, averaged over non-carcinoma cancer types'.format(gene))
+plt.tight_layout()
+
+
+# ### Plot performance broken down by cancer type
```

Comment:
I think I got confused here. Are you doing a pairwise experiment? I assumed "all other cancer types" was all of them concatenated, not one at a time, so I don't know how the plots in chunk 12 work.

Reply:
The training set is all of them concatenated, but the test set for the "cancer type" experiments in this script is always only a single cancer type, as opposed to the "stratified" experiments from before where the test set is also all cancer types concatenated.

This plot is showing performance on the test set, so we can split that out by cancer type even though the training set is (largely) the same data.